### ISSUE TITLE
ArgoCD is ignoring Helm Release capabilities

### ISSUE THREAD

**Issue by ArieLevs** [argoproj/argo-cd#7637]
There is a similar issue https://github.com/argoproj/argo-cd/issues/7438 regarding same issue that has been closed.
Also seen a fix to the Helm capabilities in [2.2 milestone](https://github.com/argoproj/argo-cd/issues/7291), can it also include `Release` part? 

**Describe the bug**

ArgoCD is ignoring `Release` [built-in objects](https://helm.sh/docs/chart_template_guide/builtin_objects/)

**To Reproduce**

Deploy any helm template that includes 
```
{{ if .Release.IsUpgrade }}
SOMETHING HERE
{{ end }}
```

**Expected behavior**

ArgoCD should respect helm release capabilities

**Version**

```shell
v2.1.3+d855831
```

**Comment by jessesuen** [comment]
I answer the question in the discussion: https://github.com/argoproj/argo-cd/discussions/7496#discussioncomment-1609267

The short answer we cannot achieve this unless Argo CD starts remembering lifecycle information about applications.

**Comment by phanama** [comment]
@jessesuen At least can the flag be exposed in `Application` definition for users to set manually? The default should still be `.Release.IsInstall`, and users can deliberately set the flag after initial installation.

The flag `helm template --is-upgrade` itself has been available since helm v3.1:
https://github.com/helm/helm/commit/cb81b07125585309d0cf5a17ce57dcd10429d8a1
